### DISCUSSION TITLE
Using forms without name

### DISCUSSION THREAD

**Discussion by lecneri** [symfony/symfony#42971] [Q&A] [answered]
Hello,

I'm trying to create an simple api (no entities yet). But in order to make my life easier, I'm trying to user form, because I can take advantange of validations. 
So I'm playing around  with this form, but I don't know how to submit file without using form[field_name].
```php
$data = [];
$form = $this
    ->createFormBuilder($data)
    ->add('spreadsheet', FileType::class)
    ->create('') //Attention here *1
    ->getForm();
$form->submit($request->files->get('spreadsheet'));
```
So, in *1 if I replace ```create('')``` for ```create('form')``` and submit a file using ```form[spreadsheet]``` form is submited and valid. If I leave *1 empty and submit ```spreadsheet``` directly, I get ```false``` in ```$form->isValid()```. Sice I'm creating an API I don't want to force using ```form[field]``` when using POST to any route.

**Comment by ker0x** [comment]
You can use the `createNamedBuilder()` method from the `FormFactory` class with an empty name.

Inside a controller:

```php
$this->container->get('form.factory')->createNamedBuilder('')
    ->add('spreadsheet', FileType::class)
    ->getForm()
;
```

**Comment by lecneri** [comment]
Hi @ker0x ,
Thank you for your reply. I did as you told, and the result it's the same as I described.

**Comment by lecneri** [comment, marked as answer]
Hi, just saying that I got it working.
Instead of  ```$form->submit($request->files->get('spreadsheet'));``` I used ```$form->submit($request->files);```
